pkg/extensions/extensions.go

```diff
@@ -156,53 +162,73 @@ func (h *Host) getExtensionPath(version, name, repo string) string {
 //	org: getoutreach
 //	repo: stencil-plugin
 //	name: github.com/getoutreach/stencil-plugin
-func (h *Host) downloadFromRemote(ctx context.Context, org, repo, name, version string) (string, error) {
-	ghc, err := github.NewClient(github.WithAllowUnauthenticated(), github.WithLogger(h.log))
+func (h *Host) downloadFromRemote(ctx context.Context, name, version string) (string, error) {
+	token, err := github.GetToken()
 	if err != nil {
-		return "", err
+		h.log.WithError(err).Warn("Failed to get github token, falling back to anonymous")
 	}
 
-	gh := updater.NewGithubUpdaterWithClient(ctx, ghc, org, repo)
+	repoURL := "https://" + name
 
-	var rel *gogithub.RepositoryRelease
 	if version == "" {
-		var err error
-		rel, err = gh.GetLatestVersion(ctx, "v0.0.0", false)
-		if err != nil {
-			return "", errors.Wrap(err, "failed to find latest extension version")
-		}
-	} else {
-		var err error
-		rel, err = gh.GetRelease(ctx, version)
+		v, err := resolver.Resolve(ctx, token, &resolver.Criteria{
+			URL: repoURL,
+		})
 		if err != nil {
-			return "", errors.Wrapf(err, "failed to find extension version %q", version)
+			return "", errors.Wrap(err, "failed to get latest version")
 		}
+		version = v.Tag
 	}
 
-	// Check if the version we're pulling already exists and is exectuable before downloading
+	// Check if the version we're pulling already exists and is executable before downloading
 	// it again.
-	dlPath := h.getExtensionPath(rel.GetTagName(), name, repo)
+	dlPath := h.getExtensionPath(version, name)
 	if info, err := os.Stat(dlPath); err == nil && info.Mode() == 0o755 {
 		return dlPath, nil
 	}
 
-	// Binary for plugin at version we want doesn't exist on disk, need to download.
-	bin, cleanup, err := gh.DownloadRelease(ctx, rel, repo, repo)
+	h.log.WithField("version", version).WithField("repo", repoURL).Debug("Downloading native extension")
+	a, archiveName, _, err := release.Fetch(ctx, token, &release.FetchOptions{
+		AssetName: filepath.Base(name) + "_" + strings.TrimPrefix(version, "v") + "_" + runtime.GOOS + "_" + runtime.GOARCH + ".tar.gz",
```

Comment:
This would probably be more readable as a `fmt.Sprintf`.

Although - is there not a shared function to build something like this yet? It seems like there should be one somewhere already.

Reply:
Yeah, this is something I've recognized as something I need to build out on the new updater rewrite I did. It's been in the back of my head for a bit, let me just go ahead and create a ticket so I don't forget and we do the work soon

Reply:
Tracking that in [DT-2786]